src/datasets/builder.py

```diff
@@ -208,6 +208,7 @@ def __init__(
         name=None,
         hash=None,
         features=None,
+        keep_in_memory=False,
```

Comment:
The role of a dataset builder is to write the dataset in the cache using `download_and_prepare`, and then return the loaded dataset from disk using `as_dataset`.

I think it would make more sense to have the `keep_in_memory` parameter directly in `as_dataset`.

What do you think ?

Reply:
@lhoestq well seen! I totally agree with you. This will eventually help for further decoupling... ;)